pkg/config/manifest.go

```diff
@@ -566,220 +568,73 @@ func (s *Step) GetMixinName() string {
 	return mixinName
 }
 
-// ResolveStep will walk through the Step's data and resolve any placeholder
-// data using the definitions in the manifest, like parameters or credentials.
-func (m *Manifest) ResolveStep(step *Step) error {
-	return reflectwalk.Walk(step, m)
-}
+func (m *Manifest) buildSourceData() (map[string]interface{}, error) {
+	data := make(map[string]interface{})
+	m.sensitiveValues = []string{}
+	bundle := make(map[string]interface{})
+	data["bundle"] = bundle
 
-// Map is a NOOP but implements the github.com/mitchellh/reflectwalk MapWalker interface
-func (m *Manifest) Map(val reflect.Value) error {
-	return nil
-}
-
-// MapElem implements the github.com/mitchellh/reflectwalk MapWalker interface and handles
-// individual map elements. It will resolve source references to their value within a
-// porter bundle and replace the value
-func (m *Manifest) MapElem(mp, k, v reflect.Value) error {
-	if v.Kind() == reflect.Interface {
-		v = v.Elem()
-	}
-	// If the value is is a map, check to see if it's a
-	// single entry map with the key "source".
-	if kind := v.Kind(); kind == reflect.Map {
-		if len(v.MapKeys()) == 1 {
-			sk := v.MapKeys()[0]
-			if sk.Kind() == reflect.Interface {
-				sk = sk.Elem()
-			}
-			//if the key is a string, and the string is source, then we should try
-			//and replace this
-			if sk.Kind() == reflect.String && sk.String() == "source" {
-				kv := v.MapIndex(sk)
-				if kv.Kind() == reflect.Interface {
-					kv = kv.Elem()
-					value := kv.String()
-					replacement, err := m.resolveValue(value)
-					if err != nil {
-						return errors.Wrap(err, fmt.Sprintf("unable to set value for %s", k.String()))
-					}
-					mp.SetMapIndex(k, reflect.ValueOf(replacement))
-				}
-			}
+	params := make(map[string]interface{})
+	bundle["parameters"] = params
+	for _, param := range m.Parameters {
+		//pe := strings.ToUpper(param.Name)
+		pe := param.Name
+		var val string
+		val, err := resolveParameter(param)
+		if err != nil {
+			return nil, err
+		}
+		if param.Sensitive {
+			m.sensitiveValues = append(m.sensitiveValues, val)
```

Comment:
This isn't on you, but the sensitive values stuff is in desperate need of a refactor so that it's not tied to the manifest like this. I can't wait for your work to split out the runtime from the manifest as I believe that's something we identified as being helpful in order to do the masking in a less obtrusive way going forward. 🤞